packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -534,6 +541,91 @@ export class CdkToolkit {
     return stacks;
   }
 
+  private stackHasNestedStacks(rootStack: cxapi.CloudFormationStackArtifact) {
+    for (const resourceName in rootStack.template.Resources) {
+      if (rootStack.template.Resources[resourceName].Type === 'AWS::CloudFormation::Stack') {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private async findPhysicalNameForNestedStack(logicalId: string, parentStackName: string | undefined, sdk: ISDK): Promise<string | undefined> {
+    if (!parentStackName) {
+      return undefined;
+    }
+    const stackResources = await this.listNestedStackResources(parentStackName, sdk);
```

Comment:
This is not cached anywhere, which means we will repeat this call unnecessarily for every Nested Stack in our parent Stack.

We need to add caching for this, and unit test it, similarly like we do for hotswapping.